Lib/uuid.py

```diff
@@ -719,14 +720,36 @@ def uuid5(namespace, name):
     hash = sha1(namespace.bytes + name).digest()
     return UUID(bytes=hash[:16], version=5)
 
+def uuid8(a=None, b=None, c=None):
+    """Generate a UUID from three custom blocks.
+    'a' is the first 48-bit chunk of the UUID (octets 0-5);
+    'b' is the mid 12-bit chunk (octets 6-7);
+    'c' is the last 62-bit chunk (octets 8-15).
+    When a value is not specified, a random value is generated.
+    """
+    if a is None:
```

Comment:
```suggestion
    if a is None and b is None and c is None:
         import random
         return UUID(int=random.getrandbits(48 + 12 + 62), version=8)

        if a is None:
```
If performance is important adding a fast path for (what I expect to be) the common case is possible. On a quick test (python 3.12) this makes uuid8 about 30 to 40 percent faster.

Reply:
As you observed, UUID8 is mainly for custom chunks so I doubt it will be used in production. Not specifying chunks in UUID8 is essentially the same as asking for a UUID4 but with a different version number. I'll keep the suggestion in a back of my head but I don't think I'll use it.